### PR TITLE
Fixes #5176 - Adding vSphere as an OS type

### PR DESCRIPTION
**Pull request by LiamLeane**
Adds simple vCenter OS family.

Templates heading to theforeman/community-templates#249

### CODE REVIEW

**Comment by mmoll** [comment]
I guess there will be a potential license problem with the logo... Also, shouldn't it be "ESXi" instead of "vCenter"?

**Comment by LiamLeane** [comment]
No but it should be vSphere, changed. Same OS type can be used to build out vCenter servers which must have been what I was thinking about when naming, also plan to use this for forthcoming OVF deployment plugin for deploying VCSA instances.

As for the logo I can't find logo release on the VMWare website so I may need to knock something up. I have seen the vSphere logo used in other OSS products so my googlefoo may simply not be sufficient to find the release.

**Comment by mmoll** [comment]
I don't quite follow on the naming... https://www.vmware.com/products/vsphere-hypervisor - it sounds to me "VMware ESXi" is called "vSphere Hypervisor" now, but "vCenter" is still "vCenter (server)" https://www.vmware.com/products/vcenter-server

**Comment by LiamLeane** [comment]
vSphere is the product suite which includes vSphere Hypervisor (or vSphere ESXi Hypervisor or vSphere ESXi depending on which part of the VMWare documentation you look at) and vCenter server.

**Comment by mmoll** [comment]
:see_no_evil:

**Comment by ekohl** [comment]
What's the licensing on the image?

**Comment by LiamLeane** [comment]
CC BY 3.0. I have a message in to the author to see if they will re-license under GPL.

Someone more familiar with IP stuff would need to weigh in on if its similarity to the real vSphere logo is a problem or not, I know VMWare usually don't have a problem with use of their logos like this but I can't find a release for the vSphere logo like with their main corporate logo.

**Comment by mmoll** [comment]
https://github.com/lzap/foreman-dummy-icons would be the place we're taking icons from otherwise

File: lib/foreman/renderer.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module Renderer
 
     ALLOWED_VARIABLES ||= [:arch, :host, :osver, :mediapath, :mediaserver, :static,
                            :repos, :dynamic, :kernel, :initrd,
-                           :preseed_server, :preseed_path, :provisioning_type]
+                           :preseed_server, :preseed_path, :provisioning_type, :bootcfg]
```
Comment (domcleal) [inline lib/foreman/renderer.rb]: I don't think this is needed, there's no `bootcfg` variable defined as `pxe_type` is kickstart, so it uses kickstart_attributes (only initrd/kernel).

File: app/models/operatingsystems/Vsphere.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Vsphere < Operatingsystem
+  PXEFILES = {:kernel => "mboot.c32", :bootcfg => "boot.cfg", :initrd => ""}
+
+  class << self
+    delegate :model_name, :to => :superclass
+  end
+
+  def mediumpath(host)
+	  medium_uri(host).to_s
```
Comment (domcleal) [inline app/models/operatingsystems/Vsphere.rb]: Nitpick, too much whitespace at the start of this line - it should have four spaces.

File: app/models/operatingsystem.rb [hunk 3]
```diff
@@ -284,4 +285,4 @@ def reject_empty_provisioning_template(attributes)
     attributes.merge!({:_destroy => 1}) if template_exists && provisioning_template_id_empty
     (!template_exists && provisioning_template_id_empty)
   end
-end
+end
```
Comment (domcleal) [inline app/models/operatingsystem.rb]: Nitpick, please keep the \n here, your editor might be configurable to keep EOLs.

File: app/helpers/operatingsystems_helper.rb [hunk 4]
```diff
@@ -53,3 +55,4 @@ def os_habtm_family(type, obj)
     result.empty? ? type : result
   end
 end
+
```
Comment (domcleal) [inline app/helpers/operatingsystems_helper.rb]: Nitpick, remove the extra line.

**Comment by domcleal** [comment]
I think the image is probably too similar and I'd prefer that we shipped something generic unless we can confirm it's either not trademarked (unlikely!) or it's permitted usage.  Though the copyright aspect's probably fixed now you've redrawn it, it's similar.

The inline comment about ALLOWED_VARIABLES should fix the merge error, so please rebase the PR against develop once that's changed.

**Comment by timogoebel** [comment]
What is the status here? I think, vSphere would be a great addition for our os family.

**Comment by dLobatog** [comment]
ping @LiamLeane

**Comment by LiamLeane** [comment]
Ill jump back on this next week.

**Comment by mmoll** [comment]
@LiamLeane  will this get opened again? :)

**Comment by mmoll** [comment]
@LiamLeane  will this get opened again? :)

**Comment by LiamLeane** [comment]
Yes, easier to just git reset them to try and merege the changes since I
last touched it.

**Pull request closed without merging**